hadoop-ozone/dist/src/main/compose/upgrade/generate-data.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script can be run only if the cluster is already started 
+# one, and initialized (but not data is written, yet).
+
+docker-compose run scm ozone freon cgscm -n 1
```

Comment:
I am not able to run this script. If I run it without creating the cluster, it fails while creating OM.
```
2021-02-23 14:03:21,345 [main] ERROR om.OzoneManager: Could not initialize OM version file
java.io.IOException: Execution of task OM#getScmInfo failed permanently after 10 attempts
	at org.apache.hadoop.hdds.utils.RetriableTask.call(RetriableTask.java:75)
	at org.apache.hadoop.ozone.om.OzoneManager.getScmInfo(OzoneManager.java:1062)
	at org.apache.hadoop.ozone.om.OzoneManager.omInit(OzoneManager.java:976)
	at org.apache.hadoop.ozone.om.OzoneManagerStarter$OMStarterHelper.init(OzoneManagerStarter.java:131)
	at org.apache.hadoop.ozone.om.OzoneManagerStarter.initOm(OzoneManagerStarter.java:95)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1952)
	at picocli.CommandLine.access$1100(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:2152)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:2530)
	at picocli.CommandLine.parseWithHandler(CommandLine.java:2465)
	at org.apache.hadoop.hdds.cli.GenericCli.execute(GenericCli.java:96)
	at org.apache.hadoop.hdds.cli.GenericCli.run(GenericCli.java:87)
	at org.apache.hadoop.ozone.om.OzoneManagerStarter.main(OzoneManagerStarter.java:51)
```
After manually starting cluster also it fails.
```
Creating upgrade_scm_run ... done
Error response from daemon: Address already in use
Creating upgrade_om_run ... done
Error response from daemon: Address already in use
Creating upgrade_dn1_run ... done
Error response from daemon: Address already in use
Creating upgrade_dn2_run ... done
Error response from daemon: Address already in use
Creating upgrade_dn2_run ... done
Error response from daemon: Address already in use
```

Reply:
> I am not able to run this script. If I run it without creating the cluster, it fails while creating OM.

Can you please share which command did you use exactly? The exception seems to be related to the start of OM which this script is running only freon tests.

Also, my version from docker-compose:
```
docker-compose version      
docker-compose version 1.28.2, build unknown
docker-py version: 4.4.1
CPython version: 3.9.1
OpenSSL version: OpenSSL 1.1.1i  8 Dec 2020
```

Can you please share your version?

I updated the script to make sure that the cluster is stopped...

Reply:
```
docker-compose version 1.27.2, build 18f557f9
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020```

I had run generate-data.sh from compose/upgrade. The first error occurred without running docker-compose up -d.
The second error occurred when running the script after running docker-compose up -d.

Reply:
What was the first and second error. I see only one exception here from OM, not from the code which is added by this patch.

Please note that this tool is offline. You should have an initialized cluster, but you should *stop* all the components during the generation. I added an implicit `docker-compose stop` step to the script to make it clear.

Reply:
I do not really have strong opinion on the script. Just wanted to try it out. But somehow I am still not able to make it work.
```
ljain-mbp15:upgrade ljain$ ./generate-data.sh
Stopping upgrade_s3g_1 ... done
Stopping upgrade_dn1_1 ... done
Stopping upgrade_scm_1 ... done
Stopping upgrade_dn3_1 ... done
Stopping upgrade_om_1  ... done
Stopping upgrade_dn2_1 ... done
Creating upgrade_scm_run ... done
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: exec: ozone: not found
ERROR: 127
Creating upgrade_om_run ... done
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: exec: ozone: not found
ERROR: 127
Creating upgrade_dn1_run ... done
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: exec: ozone: not found
ERROR: 127
Creating upgrade_dn2_run ... done
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: exec: ozone: not found
ERROR: 127
Creating upgrade_dn2_run ... done
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: exec: ozone: not found
ERROR: 127
```
I tried changing the script in this manner to make it work. But it still failed for me.
```
docker-compose run scm /opt/hadoop/bin/ozone freon cgscm -n 1
```
```
Creating upgrade_scm_run ... done
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: /opt/hadoop/bin/ozone: No such file or directory
ERROR: 127
```

Reply:
```
To use Ozone please mount ozone folder to /opt/hadoop
/usr/local/bin/entrypoint.sh: line 22: exec: ozone: not found
```

Looks like an independent error. The mounts are not working. Either because you did a clean and build since the last container run or because you started it from the source directories.

I would be happy to help to debug it, just ping me on slack....